### PR TITLE
add weighting to interface of pushers

### PR DESCRIPTION
**Pull request by PrometheusPi**
This pull request follows #1201. It **only** adds the weighting to the pusher interface.

### CODE REVIEW

**Comment by PrometheusPi** [comment]
Please wait with merging till I have access to hypnos again to fix the license header.

**Comment by PrometheusPi** [comment]
done with license update

File: src/picongpu/include/particles/pusher/particlePusherPhot.hpp [hunk 1]
```diff
@@ -33,14 +33,15 @@ namespace picongpu
...
-                     typename T_Charge>
+                     typename T_Charge,  typename T_Weighting>
                     __host__ DINLINE void operator()(
                                                         const T_Bfield ,
                                                         const T_Efield ,
                                                         T_Pos& pos,
                                                         T_Mom& mom,
                                                         const T_Mass ,
-                                                        const T_Charge )
+                                                        const T_Charge ,
```
Comment (psychocoderHPC) [inline src/picongpu/include/particles/pusher/particlePusherPhot.hpp]: Could you please remove the `space` after the variable name. Please do the some for other pushers. I missed that you add `space`'s in issue #1210

Comment (PrometheusPi) [inline src/picongpu/include/particles/pusher/particlePusherPhot.hpp]: done

**Pull request merged**